chain/client-primitives/src/types.rs

```diff
@@ -33,6 +33,7 @@ pub enum Error {
     Chunk(near_chunks_primitives::Error),
     BlockProducer(String),
     ChunkProducer(String),
+    TransactionPoolFull,
```

Comment:
I think this error should be added to near_chain_primitives::Error, since the TransactionPool struct sits in chain (whether that should be changed is another question). This will eliminate the need to add a dependency to client-pritimitives in chain, which feels a little weird.